tensorboard/backend/application.py

```diff
@@ -174,6 +175,7 @@ def __init__(self, plugins):
         plugin_apps = plugin.get_plugin_apps()
       except Exception as e:  # pylint: disable=broad-except
         if type(plugin) is core_plugin.CorePlugin:  # pylint: disable=unidiomatic-typecheck
+          six.reraise(*sys.exc_info())
```

Comment:
(a) Using `six.reraise` is better than `raise e`, but shouldn't we remove the following `raise e`?
(b) You can just `raise` to preserve the original stack trace:
```py
import six
import sys
import threading

def a():
  b()
def b():
  c()
def c():
  raise AssertionError("oh, dear")

def try_with_raise_exception():
  try:
    a()
  except Exception as e:
    raise e

def try_with_six_reraise():
  try:
    a()
  except Exception:
    six.reraise(*sys.exc_info())

def try_with_bare_raise():
  try:
    a()
  except Exception:
    raise

def test(f):
  t = threading.Thread(target=f, name=f.__name__)
  t.start()
  t.join()

if __name__ == '__main__':
  test(try_with_raise_exception)  # does not preserve stack trace
  test(try_with_six_reraise)      # preserves stack trace
  test(try_with_bare_raise)       # preserves stack trace
```